Spokestack/WebRTCVAD.swift

```diff
@@ -137,14 +153,20 @@ private var frameBuffer: RingBuffer<Int16>!
                     }
                 }
             }
-            if detected {
-                if !self.context.isSpeech {
-                    self.context.isSpeech = detected
-                }
-            } else {
-                if self.context.isSpeech {
-                    self.context.isSpeech = detected
-                }
+            // if speech activity is already detected, continue until the minimum detection length is reached
+            if self.context.isSpeech && self.detectionLength <= self.minDetectionLength {
+                self.detectionLength += 1
+            // if speech activity is detected, continue until the maximum detection length is reached
+            } else if detected && self.detectionLength > 0 && self.detectionLength <= self.maxDetectionLength {
```

Comment:
Not sure you need the first check on `detectionLength` here.

Also, you check the max here, but nowhere else—it's not acting as an actual max activation, though I'm not sure whether we'd want one in the VAD at all.

Reply:
The first `detectionLength` check is in order to skip this case if it's a new detection.

The `maxDetectionLength` check is done here because that will reject an excessive `detectionLength` and fall through to the edge case.

Reply:
If `detected` is `true`, the edge won't be triggered; nothing will happen. I think maybe what you want for that last check is `if !detected && self.detectionLength > self.minDetectionLength`, then you can get rid of the max altogether. As Brent pointed out, this runs a bit counter to the logic in Android, which requires a certain number of negative samples to flip rather than relying on a single oscillation past the minimum.

Reply:
> If detected is true, the edge won't be triggered;
Yes, and so we allow the activation to continue (which isn't nothing, the activation counter increments). Your proposal to remove max will result in the counter never getting reset. 

I think at this point we can agree that the behavior is consistent with the released behavior. Since this refactor does seem to fix a bug, I will defer using a negative sample counter to the future.

Reply:
`if detected && self.context.isSpeech && self.detectionLength > self.maxDetectionLength`
If those conditions occur together, nothing changes, and nothing gets incremented.

Like you said, though, if the bug it's meant to fix is fixed, release now and refactor later.